fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/handler/data/ElectionHandler.java

```diff
@@ -178,13 +185,25 @@ public void handleElectionEnd(HandlerContext context, ElectionEnd electionEnd)
    * @param castVote the message that was received
    */
   public void handleCastVote(HandlerContext context, CastVote castVote)
-      throws UnknownElectionException, DataHandlingException {
+      throws UnknownElectionException, DataHandlingException, UnknownLaoException {
     Channel channel = context.getChannel();
     MessageID messageId = context.getMessageId();
     PublicKey senderPk = context.getSenderPk();
 
     Timber.tag(TAG).d("handleCastVote: channel %s", channel);
+
+    String laoId = castVote.getLaoId();
+    if (!laoRepo.containsLao(laoId)) {
+      throw new UnknownLaoException(laoId);
+    }
+
+    // Election id validity is checked with this
     Election election = electionRepository.getElectionByChannel(channel);
+
+    if (election.getCreation() > castVote.getCreation()) {
+      throw new DataHandlingException(castVote, "vote cannot be older than election creation");
```

Comment:
Here vote and election creation are switched

Reply:
Are you sure? I want to check that the election creation comes before the vote creation.

Reply:
Yes the `if` condition is right, I was referring to the error string

Reply:
I think both are correct, the error is thrown if the vote is older than the election creation.